### PR TITLE
Feat/randomize

### PR DESCRIPTION
**Pull request by grandq33769**
* [修正] 每天隨機打卡
* [新增] 依據上班時間，在合法時間內進行隨機下班打卡
* [新增] 依據假期日，跳過打卡日，並在月尾儲存下一個月的假期

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,29 +44,50 @@ const login = async (page, url, id, pw) => {
...
-  await delay(3000);
-  const text = await page.evaluate(() => Array.from(document.querySelectorAll('table:nth-child(2) :nth-child(1) td[bgcolor="#FFCCCC"]'), element => element.textContent));
-  const holidays = text.map(Number);
+
+
+  var today = new Date(), y = today.getFullYear(), m = today.getMonth();
+  var lastDay = new Date(y, m + 1, 0);
+
+  // 如果今天是這個月的最後一天，把下個月的假期存起來
+  if(today.getTime() == lastDay.getTime()) {
```
Comment (chihsuan) [inline index.js:63]: Did you test this? It seems that you compare to "time".

Should it be like this?
https://stackoverflow.com/questions/6355063/how-to-validate-date-if-is-the-last-day-of-the-month-with-javascript

**Comment by chihsuan** [comment]
Could you replace `var` with `const` or `let`?   `var` is used before JS ES5 version (not recommend).

File: index.js (hunk 1 above)
Comment (grandq33769) [reply index.js:63]: I think I should change `==` operator `===`. What do you think ?

[Get Last Day](https://stackoverflow.com/questions/13571700/get-first-and-last-date-of-current-month-with-javascript-or-jquery/13572682)

[Determine the Date are equal](https://stackoverflow.com/questions/492994/compare-two-dates-with-javascript/497790)

Comment (chihsuan) [reply index.js:63]: In JS, Date object just like Python's datetime includes time so I think this won't work.

for example:
```
> new Date()
< Fri Aug 21 2020 13:26:31 GMT+0800 (CST)
> new Date('2020', 9, 0)
< Wed Sep 30 2020 00:00:00 GMT+0800 (CST)
```

Comment (chihsuan) [reply index.js:63]: The correct way might be like this

```js
function isLastDay(dt) {
    var test = new Date(dt.getTime()),
        month = test.getMonth();

    test.setDate(test.getDate() + 1);
    return test.getMonth() !== month;
}
```

Comment (grandq33769) [reply index.js]: Yes, you are right. I have tested it and fix it in 205907c93ac77d0ed20055c88db2138bc03aa32e. Thanks a lot ~

**Comment by grandq33769** [comment]
> Could you replace `var` with `const` or `let`? `var` is used before JS ES5 version (not recommend).

No problem. Fixed in db3f932db99f562d13842114fdcbb8844b7aa3c4.

**Comment by chihsuan** [comment]
LGTM.

**Pull request merged**